gems/aws-sdk-core/lib/aws-sdk-core/instance_profile_credentials.rb

```diff
@@ -5,145 +5,139 @@
 
 module Aws
   # An auto-refreshing credential provider that loads credentials from
-  # EC2 instances.
+  # EC2 instances using IMDSv2.
   #
-  #     instance_credentials = Aws::InstanceProfileCredentials.new
-  #     ec2 = Aws::EC2::Client.new(credentials: instance_credentials)
+  #     ec2_metadata = Aws::EC2Metadata.new # with customized opts
+  #     instance_creds = Aws::InstanceProfileCredentials.new(
+  #       client: ec2_metadata
+  #     )
+  #     ec2_client = Aws::EC2::Client.new(credentials: instance_creds)
+  #
+  # If you omit the `:client` option, a new {Aws::EC2Metadata} will
+  # be created with options provided.
```

Comment:
I probably should expand on the documentation regarding the two different apis and the how the flow goes. not sure if there will be a upstream doc that will handle that instead